bin/lesson_check.py

```diff
@@ -257,37 +257,42 @@ def read_all_markdown(source_dir, parser):
     {path : {'metadata':yaml, 'metadata_len':N, 'text':text, 'lines':[(i, line, len)], 'doc':doc}}
     """
 
-    all_dirs = [os.path.join(source_dir, d) for d in SOURCE_DIRS]
-    all_patterns = [os.path.join(d, '*.md') for d in all_dirs]
     result = {}
-    for pat in all_patterns:
-        for filename in glob.glob(pat):
+    for d in SOURCE_DIRS:
+        dpath = os.path.join(source_dir, d)
+
+        pattern = os.path.join(dpath, '*.md')
+        for filename in glob.glob(pattern):
             data = read_markdown(parser, filename)
             if data:
                 result[filename] = data
+
```

Comment:
this change looks good but it not related to Win10 compatibility. Let's do this in a different PR.